java/flight/flight-core/pom.xml

```diff
@@ -175,6 +177,13 @@ under the License.
                   <include>com.google.protobuf:*</include>
                 </includes>
               </artifactSet>
+              <filters>
+                <filter>
+                  <excludes>
+                    <exclude>**/module-info.class</exclude>
```

Comment:
Does this save us from "module hell"? aka publishing a package that has automatic module names auto-generated from filenames?

e.g.
```
[WARNING] * Required filename-based automodules detected: [flatbuffers-java-24.3.25.jar, jsr305-3.0.2.jar, protobuf-java-3.25.1.jar, protobuf-java-util-3.25.1.jar]. Please don't publish this project to a public artifact repository! *
```

Reply:
Not sure., maybe we can add [`Automatic-Module-Name` attribute](https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/lang/module/ModuleFinder.html) in the shaded jars to make the name not filename dependent. 

What it helps with though is having shaded jars which exposes the wrong module information (like the jackson or gson ones which are copied over under `META-INF/versions/9/module-info.class`)

Reply:
FWIW I did request flatbuffers to add the Automatic-Module-Name attribute: https://github.com/google/flatbuffers/issues/8348

Reply:
There's no built-in way to merge all `module-info.class` files into a single one, one need to analyze what needs to be exported and how to deal also with relocation.

Specifically for this module, there are 3 different shaded jars with different set of dependencies included and different set of relocations and I'm not sure which ones are supposed to be used internally and which ones are to be used externally. Knowing the intended usage may also guide us in deciding if those artifacts should be modularized or not

Reply:
Maybe I should open a separate issue/pull request? I tried to analyze how many shaded artifacts are generated and how many are used.

Here's the list of shaded artifacts:
- arrow-tools-{version}-jar-with-dependencies.jar
- flight-sql-jdbc-driver-{version}.jar
- flight-integration-tests-{version}-jar-with-dependencies.jar
- flight-core-{version}-shaded.jar
- flight-core-{version}-shaded-ext.jar
- flight-core-{version}-jar-with-dependencies.jar
- arrow-performance-{version}-benchmarks.jar
- arrow-vector-{version}-shade-format-flatbuffers

`arrow-vector` is used by other modules to not expose `flatbuffers` directly. 

I can also see `arrow-tools`/`flight-integration-tests` used by some tests defined in `dev/tasks.yml`, and `arrow-perfornance` uber jar is common for people who wants to run jmh from commandline:  those shaded jars are useful and I do not expect that we need modularization for those jars

The jdbc driver is shaded because people expect it to be self contained. It could benefit from having its own module-info.java which would only allow exposes the driver + public api, but all relocated classes would be not accessible.

Finally there are all the `flight-core` shaded jars and I haven't figured out if they are used or not

Reply:
I think a separate pull request for shaded artifacts is okay. We should also get rid of the warning I pasted initially before releasing v18. Someone on the original GH issue recommended using shading to prevent us from exposing those module names. Maybe that can be a separate issue, too.